src/pandroid/app/src/main/java/com/panda3ds/pandroid/data/config/GlobalConfig.java

```diff
@@ -0,0 +1,61 @@
+package com.panda3ds.pandroid.data.config;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.panda3ds.pandroid.app.PandroidApplication;
+import com.panda3ds.pandroid.utils.Constants;
+
+import java.io.Serializable;
+
+public class GlobalConfig {
+    private static SharedPreferences data;
+
+    public static void initialize() {
+        data = PandroidApplication.getAppContext()
+                .getSharedPreferences(Constants.PREF_GLOBAL_CONFIG, Context.MODE_PRIVATE);
+    }
+
+    private static <T extends Serializable> T get(Key<T> key) {
+        Serializable value;
+        if (key.defValue instanceof String) {
+            value = data.getString(key.name, (String) key.defValue);
+        } else if (key.defValue instanceof Integer) {
+            value = data.getInt(key.name, (int) key.defValue);
+        } else if (key.defValue instanceof Boolean) {
+            value = data.getBoolean(key.name, (Boolean) key.defValue);
+        } else if (key.defValue instanceof Long) {
+            value = data.getLong(key.name, (Long) key.defValue);
+        } else {
+            value = data.getFloat(key.name, ((Number) key.defValue).floatValue());
```

Comment:
Assuming my other comment is true that we can use primitive types, is there really a need to cast to `Number` first? can't we just `(float)key.defaultValue`?

Reply:
this really not necessary i'll remove this